articles/react-native-firebase-storage/index.md

```diff
@@ -0,0 +1,605 @@
+
+![Storage Rules Tab](/engineering-education/react-native-firebase-storage/storage_default_rules.png)
+
+Now, replace the existing rules with this.
+
+```rules
+rules_version = '2';
+service firebase.storage {
+  match /b/{bucket}/o {
+    match /{allPaths=**} {
+      allow read, write: if true;
+    }
+  }
+}
+```
+
+If you'd like to learn more about cloud storage rules, refer [here](https://firebase.google.com/docs/storage/security).
+
+### Building the UI
+In the `App.js`, let's add 4 buttons to the screen.
+
+- Take a Photo.
+
+- Record Video.
+
+- Pick a Photo.
+
+- Pick a Video.
+
+```JSX
+import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
+
+export default function App() {
+  return (
+    <View style={styles.screen}>
+      <Text style={styles.title}>Firebase Storage</Text>
+      <View>
+        <TouchableOpacity style={styles.button}>
+          <Text style={styles.buttonText}>Take Photo</Text>
+        </TouchableOpacity>
+        <TouchableOpacity style={styles.button}>
+          <Text style={styles.buttonText}>Record Video</Text>
+        </TouchableOpacity>
+        <TouchableOpacity style={styles.button}>
+          <Text style={styles.buttonText}>Pick a Photo</Text>
+        </TouchableOpacity>
+        <TouchableOpacity style={styles.button}>
+          <Text style={styles.buttonText}>Pick a Video</Text>
+        </TouchableOpacity>
+      </View>
+    </View>
+  );
+}
+```
+
+Styles:
+
+```JSX
+const styles = StyleSheet.create({
+  screen: {
+    flex: 1,
+    alignItems: 'center',
+  },
+  title: {
+    fontSize: 35,
+    marginVertical: 40,
+  },
+  button: {
+    backgroundColor: '#47477b',
+    color: '#fff',
+    justifyContent: 'center',
+    alignItems: 'center',
+    paddingVertical: 15,
+    paddingHorizontal: 40,
+    borderRadius: 50,
+    marginTop: 20,
+  },
+  buttonText: {
+    color: '#fff',
+  },
+});
+```
+
+![App UI](/engineering-education/react-native-firebase-storage/app_ui.jpg)
+
+### Adding media picker
+Now, the first 2 buttons should open the camera to take a photo and record a video respectively, and the next 2 buttons should open the gallery to pick an image and video respectively.
+
+Let's install the `react-native-image-picker` to add these functionalities.
+
+```bash
+npm install react-native-image-picker
+```
+
+> The minimum target SDK for the React Native Image Picker is 21. If your project targets an SDK below 21, bump up the minSDK target in `android/build.gradle`.
+
+After the package is installed, import the `launchCamera` and `launchImageLibrary` functions from the package.
+
+```JSX
+import { launchCamera, launchImageLibrary } from 'react-native-image-picker';
+```
+
+Both functions accept 2 arguments. The first argument is `options` for the camera or the gallery, and the second argument is a callback function. This callback function is called when the user picks a media or cancels the operation.
```

Comment:
Could you talk more on what the function returns to determine the user's action or choice.

Reply:
The value that the function returns is not important. I think those functions return nothing. That's why it's accepting callback functions. 

Discussing the return value of those functions provide no value to the topic.